pgvectorscale/src/access_method/upgrade_test.rs

```diff
@@ -28,8 +29,8 @@ pub mod tests {
         extname: &str,
         amname: &str,
     ) {
-        if cfg!(feature = "pg17") {
-            // PG17 is only supported for one version
+        if cfg!(feature = "pg17") && version != "0.4.0" {
```

Comment:
This will break when the next version is released - this should write out all previous versions, or use [`semver`](https://crates.io/crates/semver) to do a proper version comparision.